File: Data Structure & Algorithms/1. Array/5. Single Number/solution.go
package main

import "fmt"

func main() {
	fmt.Println(singleNumber([]int{2, 2, 1}))
	fmt.Println(singleNumber([]int{4, 1, 2, 1, 2}))
	fmt.Println(singleNumber([]int{1}))
}

func singleNumber(nums []int) int {
	a := 0

	for _, num := range nums {
		a ^= num
	}

	return a

}

File: Data Structure & Algorithms/1. Array/5. Single Number/Solution.java
class Solution {
    public int singleNumber(int[] nums) {
        int ans = 0;

        for(int num : nums) {
            ans ^= num;
        }

        return ans;
    }

    public static void main(String[] args) {
        Solution sol = new Solution();

        int[] nums = new int[]{2, 2, 1};
        System.out.println(sol.singleNumber(nums));

        nums = new int[]{4, 1, 2, 1, 2};
        System.out.println(sol.singleNumber(nums));   

        nums = new int[]{1};
        System.out.println(sol.singleNumber(nums));   
    }
}